docs/ROADMAP.md

```diff
@@ -7,38 +7,54 @@ category: Roadmap
 
 This document defines the roadmap for TiKV development.
 
+## Engine
+- [ ] Titan - Use the separated key-value engine in production
+- [ ] Pluggable Engine Interface - Clean up the engine wrapper code and provide more extendibility
+
 ## Raft
-- [x] Region Merge - Merge small Regions together to reduce overhead
-- [x] Local Read Thread - Process read requests in a local read thread
-- [x] Split Region in Batch - Speed up Region split for large Regions
-- [x] Raft Learner - Support Raft learner to smooth the configuration change process
-- [x] Raft Pre-vote - Support Raft pre-vote to avoid unnecessary leader election on network isolation
 - [ ] Joint Consensus - Change multi members safely.
-- [ ] Multi-thread Raftstore - Process Region Raft logic in multiple threads
-- [ ] Multi-thread apply pool - Apply Region Raft committed entries in multiple threads
+- [ ] Quiescent region - Reduce the heartbeat cost of inactive regions
+- [ ] Raft engine - Customize engine to replace RocksDB
+- [ ] Remove KV RocksDB WAL - Use Raft log as the WAL of state machine engine
+- [ ] Crossing datacenter optimization
+    - [ ] Raft Witness Role - Can only vote but not sync logs from Raft leader
+    - [ ] Follower snapshot - Get the snapshot from the follower when a new node added 
+    - [ ] Chain replication - Get Raft logs from the not leader node
 
-## Engine
-- [ ] Titan - Separate large key-values from LSM-Tree
-- [ ] Pluggable Engine Interface - Clean up the engine wrapper code and provide more extendibility
+## Distributed Transaction 
 
-## Storage
 - [ ] Flow Control - Do flow control in scheduler to avoid write stall in advance
-
-## Transaction
-- [x] Optimize transaction conflicts
-- [ ] Distributed GC - Distribute MVCC garbage collection control to TiKV
+- [ ] History version - Use another place to save history version data
+- [ ] Performance
+    - [ ] Optimize storage format 
+    - [ ] Reduce the latency of getting timestamp
+    - [ ] Optimize transaction conflicts
 
 ## Coprocessor
-- [x] Streaming - Cut large data set into small chunks to optimize memory consumption
+
+- [ ] More Pushdown expressions support
+- [ ] Common expression optimization 
 - [ ] Chunk Execution - Process data in chunk to improve performance
```

Comment:
We can separate Chunk execution into multiple tasks now:

- [ ] Batch executor - Process multiple rows at a time
- [ ] Vectorized expression evaluation - Evaluate expressions by column
- [ ] Chunk - Compute over continuous memory